apps/src/aichat/views/teacherFeedback/CleanFeedbackFooter.tsx

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames';
+import React, {memo} from 'react';
+
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import {Button, buttonColors} from '@cdo/apps/componentLibrary/button';
+import {WithTooltip} from '@cdo/apps/componentLibrary/tooltip';
+import {EmText} from '@cdo/apps/componentLibrary/typography';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {AiChatTeacherFeedback as TeacherFeedback} from '@cdo/generated-scripts/sharedConstants';
+
+import aichatI18n from '../../locale';
+import {submitTeacherFeedback} from '../../redux/aichatRedux';
+import {FeedbackValue} from '../../types';
+
+import moduleStyles from './teacher-feedback-footer.module.scss';
+
+interface Props {
+  id: number;
+  teacherFeedback?: FeedbackValue;
+  role: Role;
+}
+
+/**
+ * Teacher feedback footer displayed for messages without any profanity violations or errors.
+ */
+const CleanFeedbackFooter: React.FC<Props> = ({id, teacherFeedback, role}) => {
+  const teacherFlagged = teacherFeedback === TeacherFeedback.CLEAN_DISAGREE;
+  const dispatch = useAppDispatch();
+  const handleFlagClick = () => {
+    dispatch(
+      submitTeacherFeedback({
+        id,
+        feedback: teacherFlagged ? undefined : TeacherFeedback.CLEAN_DISAGREE,
+      })
+    );
+  };
+
+  return (
+    <div
+      className={classNames(
+        moduleStyles.teacherFeedbackContainer,
+        role === Role.ASSISTANT && moduleStyles.assistantFeedback
+      )}
+    >
+      {teacherFlagged && (
+        <EmText>{aichatI18n.chatMessage_hasBeenFlagged()}</EmText>
+      )}
+      <WithTooltip
+        key={`flag-tooltip-${teacherFlagged}`}
+        tooltipProps={{
+          tooltipId: 'flag-tooltip',
+          direction: role === Role.ASSISTANT ? 'onRight' : 'onLeft',
```

Comment:
Don't have to figure this out now necessarily, but curious how this would work in RTL languages. For now, looks really nice!